pandamium_datapack/data/pandamium/functions/misc/spawnpoint/advancement.mcfunction

```diff
@@ -0,0 +1,5 @@
+advancement revoke @s only pandamium:spawnpoint_set_via_respawn_anchor
+advancement revoke @s only pandamium:spawnpoint_set_via_bed
+scoreboard players set @s set_spawnpoint 1
+schedule function pandamium:misc/spawnpoint/advancement_ 1 append
+#only check the player nbt data at most once per tick, in case someone spam clicks their bed/respawn anchor

```

Comment:
Do I understand correctly that you split up one function into three and are using an extra scoreboard objective just to prevent players from setting their home more than once?
Is it even possible to interact with beds/anchor points more than once per tick? And even if it is, I don't think this is necessary, this shouldn't happen often by accident, and if players want to make the server lag on purpose, there are much more effective ways than spam setting your spawnpoint...
I think having everything in one function would be much cleaner

Reply:
A modified client is probably able to set the spawn point at a very high rate.
But you're right, it's very unlikely someone will do this, so i will put everything in one function.